src/classes/AFFL_MultiRecordTypeMapper.cls

```diff
@@ -136,12 +136,23 @@ public with sharing class AFFL_MultiRecordTypeMapper {
     ********************************************************************************************************/
 	public String getKeyAfflFieldByAccRecordType(ID recordTypeId) {
 		for(String recTypeName : mapAccRecTypeToContactLabel.keySet()) {
-    		ID rcId = Schema.Sobjecttype.Account.getRecordTypeInfosByName().get(recTypeName).getRecordTypeId();
-    		if(rcId == recordTypeId) {
-    			String fieldLabel = mapAccRecTypeToContactLabel.get(recTypeName);
-    			String fieldName = contactLabelNames.get(fieldLabel);
-    			return fieldName;
-    		}
+            if (Schema.Sobjecttype.Account.getRecordTypeInfosByName().get(recTypeName) != null)
+            {
+        		ID rcId = Schema.Sobjecttype.Account.getRecordTypeInfosByName().get(recTypeName).getRecordTypeId();
+        		if(rcId == recordTypeId) {
+        			String fieldLabel = mapAccRecTypeToContactLabel.get(recTypeName);
+        			String fieldName = contactLabelNames.get(fieldLabel);
+        			return fieldName;
+        		}
+            }else{
+                //TODO: Need Liane's review on the error msg
+                //Setup null pointer error message
+                String[] params = new String[]{
+                    String.valueOf(recTypeName)
+                };
+                String nullPointerMsg = String.format(label.afflNullPointerError, params);
+                Trigger.new[0].addError(nullPointerMsg);
```

Comment:
Yes, I would throw an exception instead of `addError()`. 

You are still in the loop as part of the TDTM class, however, as I reviewed this thinking through how I would approach it I realized the problem is with the configuration and not with any particular object. 

If "bulkified", every object in the batch would show the same error, and because of that, it is not an "object" issue, but an issue with the configuration. With all that in mind, if I were working on this, I would throw an exception, halting the entire transaction inside this same method. 

Since this is part of an overall transaction managed by TDTM, the TDTM Trigger Handler will deal with error handling.